src/v/container/fragmented_vector.h

```diff
@@ -81,19 +84,6 @@ class fragmented_vector {
     using pointer = T*;
     using const_pointer = const T*;
 
-    /**
-     * The maximum number of bytes per fragment as specified in
-     * as part of the type. Note that for most types, the true
-     * number of bytes in a full fragment may as low as half
-     * of this amount (+1) since the number of elements is restricted
-     * to a power of two.
-     */
-    static constexpr size_t max_frag_bytes = elems_per_frag * sizeof(T);
-
-    static_assert(
-      max_frag_bytes <= max_allocation_size,
-      "max size of a fragment must be <= 128KiB");
```

Comment:
this seemed like a nice safety guard?

Reply:
Yeah we actually have these in unit tests. I guess I can move this to the capacity method?

Reply:
Done.